spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1582,19 +1582,29 @@ def paste(self):
         text has 'CRLF' EOL chars
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
-        if clipboard.mimeData().hasUrls():
-            return
-        else:
-            text = to_text_string(clipboard.text())
-            if len(text.splitlines()) > 1:
-                eol_chars = self.get_line_separator()
-                text = eol_chars.join((text + eol_chars).splitlines())
-                clipboard.setText(text)
-            # Standard paste
-            TextEditBaseWidget.paste(self)
-            self.document_did_change(text)
+        text = to_text_string(clipboard.text())
+        # This is here to make copied files/folders to be pasted as text path.
+        # See issue 8566 and PR: 8576 for the details.
+        urls = []
+        if (not clipboard.mimeData().hasFormat('text/plain') and
+                clipboard.mimeData().hasUrls()):
+            urls = clipboard.mimeData().urls()
+            if len(urls) > 1:
+                text = ''.join('"' + url.toLocalFile().replace(osp.os.sep, '/')
+                               + '",' + '\n' for url in urls)
+            else:
+                text = urls[0].toLocalFile() + '\n'

```

Comment:
I don't think that most users would want a `'\n'` here. Usually the path will be pasted in line.

Reply:
@bcolsen , I agree to your point, I was wrong, I will remove from multiple files too. Thanks.

Reply:
in #8494 your suggested standard is followed.

Reply:
@Khalilsqu @CAM-Gerlach Should we put quotes around single lines?  It could be quite handy.

Reply:
Good thought, but IMO no, for the same reasons articulated in #8494 . Users may prefer either single or double quotes in any given context; the single path could be part of a longer string, and the user may well be pasting the path within existing quotes. Meanwhile, if the user wants single or double quotes, the appropriate one can be added with one keystroke with Spyder's auto-quote insertion turned on.

Reply:
Ok no quotes then. Auto-quote insertion works for me. :+1: